suppkg_rebuild.py

```diff
@@ -0,0 +1,255 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2017 SUSE Linux GmbH
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import argparse
+import logging
+import sys
+import urllib2
+import re
+import yaml
+from xml.etree import cElementTree as ET
+from collections import defaultdict
+
+import osc.conf
+import osc.core
+
+from osc import oscerr
+
+OPENSUSE = 'openSUSE:Factory'
+
+makeurl = osc.core.makeurl
+http_GET = osc.core.http_GET
+http_POST = osc.core.http_POST
+http_PUT = osc.core.http_PUT
+
+class StagingHelper(object):
+    def __init__(self, project):
+        self.project = project
+        self.apiurl = osc.conf.config['apiurl']
+        self.debug = osc.conf.config['debug']
+
+    def load_rebuild_data(self, project, package, filename):
+        url = makeurl(self.apiurl, ['source', project, package, '{}?expand=1'.format(filename)])
+        try: 
+            return http_GET(url)
+        except urllib2.HTTPError:
+            return None
+
+    def save_rebuild_data(self, project, package, filename, content):
+        url = makeurl(self.apiurl, ['source', project, package, filename])
+        http_PUT(url + '?comment=support+package+rebuild+update', data=content)
+
+    def rebuild_project(self, project):
+        query = {'cmd': 'rebuild'}
+        url = makeurl(self.apiurl,['build', project], query=query)
+        http_POST(url)
+
+    def get_source_packages(self, project):
+        """Return the list of packages in a project."""
+        query = {'expand': 1}
+        root = ET.parse(http_GET(makeurl(self.apiurl,['source', project],
+                                 query=query))).getroot()
+        packages = [i.get('name') for i in root.findall('entry')]
+        
+        return packages
+
+    def get_prj_pseudometa(self, project):
```

Comment:
Is there a reason `stagingapi` cannot be used for this and other already available functions?